AudioPlayer/AudioPlayer/player/extensions/AudioPlayer+PlayerEvent.swift

```diff
@@ -27,8 +27,8 @@ extension AudioPlayer {
             pausedForInterruption = true
             pause()
 
-        case .interruptionEnded where pausedForInterruption:
-            if resumeAfterInterruption {
+        case .interruptionEnded(let shouldResume) where pausedForInterruption:
+            if resumeAfterInterruption && shouldResume {
```

Comment:
Instead of checking two properties that more or less means the same, is there a possibility that shouldResume variable affects the value of resumeAfterInterruption before arriving to this point, so all the logic is centralised in one place? Usually, when I start having several boolean flags in a class, I start thinking if I can model things better with just an enum variable that gets changed when different events happen, having all the state changes occurring in one function, avoiding possible inconsistencies. Also, this helps checking for this value later, because you don't have to remember of the flags and merge them, that job is already done. What do you think?